### ISSUE TITLE
When using the article builder it appears that widgets introtext and and fulltext don't process content plugins

### ISSUE THREAD

**Issue by felixkat** [templaza/astroid-framework#892]
When using the article builder it appears that widgets introtext and and fulltext don't process content plugins but article_text does.

It would useful if text and other similiar widgets would process content plugins, maybe it could be an option to enable or disable to save processing if people don't use them.

![image](https://github.com/user-attachments/assets/2b69996c-c650-4e16-821b-b64b91fbef4a)

**Comment by sonvnn** [comment]
This looks like a problem with Joomla core. Because the introtext and fulltext variables are called from Com_content. Can you help me move this issue to Joomla git? You can create an issue here https://github.com/joomla/joomla-cms

**Comment by felixkat** [comment]
I've looked into this in a lot more detail now as I was attempting to do a non standard setup. This was mainly for a workaround or using content plugins in Astroid, (as per my post in discussions).

I've managed to find a working solution now which solves all issues.  Adding a couple of lines into the introtext element does what is required so I can add that permanently with an override.  Although if possible I might put it into an option for introtext and fulltext, i.e an option whether or not to process content plugins and have it default to no for backward compatibility.

```
// No direct access.
defined('_JEXEC') or die;

use Joomla\CMS\HTML\HTMLHelper;

extract($displayData);
$item = $options['article'];
//echo $item->article->introtext;
echo HTMLHelper::_('content.prepare', $item->article->introtext);
```

The only case I have discovered so far where this doesn't work is if the plugin content is specific as to where the content is coming from.  One plugin I am using has,

```
        $canProceed = strpos($context, 'com_content');
        if ($canProceed === false) {
            return;
        }
```

which means it doesn't run as **$context** is **'text'** when using the `echo HTMLHelper::_('content.prepare', $item->article->introtext);` code from the widget.  However commenting it out works perfect, so I will pursue a non Astroid fix for that one.  😄 


Just a couple of notes to myself and other if they stumble across this isssue. Most of the below may be common knowledge but I've detailed everything for completeness.  

If you write an article without the "read more" everything gets stored in the database table as introtext,  When you put a "read more" into the article it will put anything before the `<hr id="system-readmore">` into introtext and anything after into fulltext.  

Content plugins are usually only processed on **$article->text**.  as this is always what displays the content.  In a 'readmore' situation, Joomla peforms `$article->text = $article->introtext` and in full article mode, `$article-> text = $article->introtext . $article->fulltext`  

Using the Article Builder you have the option of displaying introtext, fulltext or text.  This is great and only becomes an issue if you want to use content plugins in either introtext or fulltext, but a small modification as shown can resolve this but it's reliant on the content plugin your are using.
